File: calc_pi.java
class calc_pi {
    public static void main(String[] args) {
        final double N = 1E7;
        double x, y, hits = 0;
        for (x = 0; x * x < N; ++x) {
            for (y = 0; y * y < N; ++y) {
                if (x * x + y * y < N) {
                    ++hits;
                }
            }
        }
        System.out.println((hits / N) * 4);
    }
}

File: calc_pi.rs
fn main() {
    const N: f64 = 1.0E7;
    let mut x: f64 = 0.0;
    let mut y: f64;
    let mut hits: f64 = 0.0;
    while x * x < N {
        y = 0.0;
        while y * y < N {
            if x * x + y * y < N {
                hits += 1.0;
            }
            y += 1.0;
        }
        x += 1.0;
    }
    println!("{}", (hits / N) * 4.0);
}
